src/core/packages/elasticsearch/server-internal/src/is_scripting_enabled.ts

```diff
@@ -8,8 +8,8 @@
  */
 
 import { defer, map, retry, timer, firstValueFrom, throwError } from 'rxjs';
+import { isRetryableEsClientError } from '@kbn/core-elasticsearch-server-utils';
```

Comment:
New behavior here is that we won't retry for `401, 403`, which makes more sense as a default behavior, like you mentioned.